### PR TITLE
X pipeline - Fix static reset with instance provided error

### PR DESCRIPTION
**Pull request by alex-vazquez**
### Purpose of this PR

It can no exist a method static named `Reset` in a class that inherits from `ScriptableObject` because it might be called during `ScriptableObject.CreateInstance` 
The reference for this method is [here](https://docs.unity3d.com/2020.1/Documentation/ScriptReference/ScriptableObject.Reset.html)

image

---
### Testing status
- URP project -> Install HDRP -> See the error

---
### Comments to reviewers
This log happened on C++ side as there is a flag that handles if the ScriptableObject that needs to be reset, so in that moment that is creating an instance is calling the method `Reset` of the object.

I found this debugging the C++ code of the Scriptable Object, but I added a new method (non-static) to the class and added some logs, to show the use behaviour here:

image

Also removed the default ctor as the code was commented.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/HDProjectSettings.cs [hunk 1]
```diff
@@ -130,15 +130,9 @@ static HDProjectSettings instance
...
-        HDProjectSettings()
-        {
-            // s_Instance.FillPresentPluginMaterialVersions(); <
-            // We can't call this here as the scriptable object will not be deserialized in yet
-        }
-
         //// We force the instance to be loaded/created and ready with valid values on assembly reload.
         [InitializeOnLoadMethod]
-        static void Reset()
+        static void InitializeFillPresentPluginMaterialVersionsOnLoad()
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/HDProjectSettings.cs:135]: @slunity When you named it Reset in https://github.com/Unity-Technologies/Graphics/commit/99384cf466fe3074e6e712f692a9191fb8f06084 , Was it to be also called when reseting the ScriptableObject from inspector contextual menu?

Comment (slunity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/HDProjectSettings.cs:135]: @RSlysz Good question :) I'm trying to remember if the naming was coincidental or not, although I wouldn't have put a static in front if it wasn't in the API. From what @alex-vazquez is saying, this is called by name match only, but more importantly, on load and/or create, and not only when the contextual menu is called? ie basically everytime the SO is created or loaded? In any case this change is ok, but I'm just wondering if there's a scenario where we should also hook reset to do this, I guess not, as the HDProjectSettings is not touched by the user it's not even in the project's Assets hierarchy, so no context menu interactions there.

Comment (RSlysz) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/HDProjectSettings.cs:135]: It was allowed to have static Reset before. Unity have changed and there is a new fence as this was causing silent issues it seams

**Review by RSlysz — APPROVED** [review]

**Review by jenniferd-unity — APPROVED** [review]

File: com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/HDProjectSettings.cs (hunk 1 above)
Comment (slunity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/HDProjectSettings.cs:135]: Ah I see. Thanks for the info!

**Review by slunity — APPROVED** [review]

**Pull request merged**